tests/test_scalars.py

```diff
@@ -0,0 +1,713 @@
+    check(assert_type(pd.Timedelta("1 nanosecond"), pd.Timedelta), pd.Timedelta)
+    check(assert_type(pd.Timedelta("1 n"), pd.Timedelta), pd.Timedelta)
+    check(assert_type(pd.Timedelta(days=1), pd.Timedelta), pd.Timedelta)
+    check(assert_type(pd.Timedelta(seconds=1), pd.Timedelta), pd.Timedelta)
+    check(assert_type(pd.Timedelta(microseconds=1), pd.Timedelta), pd.Timedelta)
+    check(assert_type(pd.Timedelta(minutes=1), pd.Timedelta), pd.Timedelta)
+    check(assert_type(pd.Timedelta(hours=1), pd.Timedelta), pd.Timedelta)
+    check(assert_type(pd.Timedelta(weeks=1), pd.Timedelta), pd.Timedelta)
+    check(assert_type(pd.Timedelta(milliseconds=1), pd.Timedelta), pd.Timedelta)
+
+    td = pd.Timedelta("1 day")
+    check(assert_type(td.value, int), int)
+    check(assert_type(td.asm8, np.timedelta64), np.timedelta64)
+
+    check(assert_type(td.days, int), int)
+    check(assert_type(td.microseconds, int), int)
+    check(assert_type(td.nanoseconds, int), int)
+    check(assert_type(td.seconds, int), int)
+    check(assert_type(td.value, int), int)
+    check(assert_type(td.resolution_string, str), str)
+    check(assert_type(td.components, Components), Components)
+
+    check(assert_type(td.ceil("D"), pd.Timedelta), pd.Timedelta)
+    check(assert_type(td.floor(Day()), pd.Timedelta), pd.Timedelta)
+    check(assert_type(td.isoformat(), str), str)
+    check(assert_type(td.round("s"), pd.Timedelta), pd.Timedelta)
+    check(assert_type(td.to_numpy(), np.timedelta64), np.timedelta64)
+    check(assert_type(td.to_pytimedelta(), dt.timedelta), dt.timedelta)
+    check(assert_type(td.to_timedelta64(), np.timedelta64), np.timedelta64)
+    check(assert_type(td.total_seconds(), float), float)
+    check(assert_type(td.view(np.int64), object), np.int64)
+    check(assert_type(td.view("i8"), object), np.int64)
+
+    check(assert_type(td + pd.Period("2012-01-01", freq="D"), pd.Period), pd.Period)
+    check(assert_type(td + pd.Timestamp("2012-01-01"), pd.Timestamp), pd.Timestamp)
+    check(assert_type(td + dt.datetime(2012, 1, 1), pd.Timestamp), pd.Timestamp)
+    check(assert_type(td + dt.date(2012, 1, 1), dt.date), dt.date)
+    check(assert_type(td + np.datetime64(1, "ns"), pd.Timestamp), pd.Timestamp)
+    check(assert_type(td + dt.timedelta(days=1), pd.Timedelta), pd.Timedelta)
+    check(assert_type(td + np.timedelta64(1, "D"), pd.Timedelta), pd.Timedelta)
+    check(
+        assert_type(
+            td + pd.period_range("2012-01-01", periods=3, freq="D"), pd.PeriodIndex
+        ),
+        pd.PeriodIndex,
+    )
+    check(
+        assert_type(td + pd.date_range("2012-01-01", periods=3), pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
+    check(
+        assert_type(td + np.array([1, 2, 3], dtype="timedelta64[D]"), np.ndarray),
+        np.ndarray,
+    )
+    check(
+        assert_type(td + np.array([1, 2, 3], dtype="datetime64[ns]"), np.ndarray),
+        np.ndarray,
+    )
```

Comment:
also test the reverse as `something + td`

Reply:
Both ways in tests.